### PR TITLE
rpm,debian,install-deps: package crimson-osd

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please.

File: ceph.spec.in [hunk 1]
```diff
@@ -650,6 +650,23 @@ ceph-osd is the object storage daemon for the Ceph distributed file
 system.  It is responsible for storing objects on a local file system
 and providing access to them over the network.
 
+%if 0%{with seastar}
+%package crimson-osd
+Summary:	Ceph Object Storage Daemon
```
Comment (b-ranto) [inline ceph.spec.in]: This would mean, we would have two packages with the same `Summary`, right?

File: ceph.spec.in [hunk 2]
```diff
@@ -650,6 +650,23 @@ ceph-osd is the object storage daemon for the Ceph distributed file
...
+%if 0%{?suse_version}
+Group:		System/Filesystems
+%endif
+Requires:	ceph-base = %{_epoch_prefix}%{version}-%{release}
+Requires:	ceph-osd = %{_epoch_prefix}%{version}-%{release}
+Requires:	lvm2
+Requires:	sudo
+Requires: libstoragemgmt
```
Comment (b-ranto) [inline ceph.spec.in]: This seems to be misaligned -- I am guessing a mix of spaces and tabs?

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [reply ceph.spec.in]: yeah. i think i should have put more background here:

the long term plan is to rename to crimson-osd to ceph-osd. and let it take the place of ceph-osd. before that happens, this packaging will only be consumed by teuthology for testing crimson-osd.

Comment (tchaikov) [reply ceph.spec.in]: @b-ranto i revised the summary to avoid the confusion in future.

File: ceph.spec.in (hunk 2 above)
Comment (tchaikov) [reply ceph.spec.in]: i removed all Requires but "ceph-osd" of "crimson-osd". since ceph-osd has all the Requires already.

**Comment by tchaikov** [comment]
@b-ranto updated and repushed.

**Comment by b-ranto** [comment]
Hmm, if ceph-osd-crimson binary is supposed to eventually supersede ceph-osd binary, wouldn't it be easier for testing it to just package the crimson binary in the `ceph-osd` package? -- I am not against this change, I am just wondering whether it wouldn't be easier to have it packaged inside the `ceph-osd` package. It would also simplify the future deprecation paths for the packages -- there would be no need for obsolets/provides in spec file, etc.

**Comment by tchaikov** [comment]
> I am just wondering whether it wouldn't be easier to have it packaged inside the `ceph-osd` package.

agreed, and i want to this in a simpler way. and i think it'd be easier for me if we can reuse the machinery of existing `# Make-Check` in `debian/control` instead of inventing a new way to messing up with the `debian/ceph-osd.install`.

> It would also simplify the future deprecation paths for the packages -- there would be no need for obsolets/provides in spec file, etc.

please note, ceph-crimson-osd package will never be published. it will only be used by our performance test performed on sepia test nodes. so, there is no need to deprecating ceph-osd. it will be a user-invisible change.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-package-crimson-osd/f03d8e254800e37c282b77a7aef244468f671903/

**Pull request merged**